packages/common/core-utils/src/binaryEncoding.ts

```diff
@@ -0,0 +1,40 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import type { JsonableOrBinary, Jsonable } from "@fluidframework/core-interfaces/internal";
+import { Uint8ArrayToString } from "@fluid-internal/client-utils";
+
+/**
+ * Encodes JsonableOrBinary into a string. Binary payloads will be base64 encoded
+ * Please note that output of this API can be used in various protocols or document format.
+ * As such, serialization format can not change. It could only be extended IF JsonableOrBinary type is extended.
+ * @param content - content to stringify
+ * @returns string
+ * @internal
+ */
+export function encodeJsonableOrBinary<T>(content: JsonableOrBinary<T>): string {
```

Comment:
@vladsud I know you closed this PR, but if/when you revisit it -- I experimented with branded strings to indicate their encoding and decoded type.  It gives some type safety when handling the serialized payload.  I'd love to get that in, either on its own or as part of the work you're doing.  I'll try to find the branch or TS Playground link.